pkg/controllers/deprovisioning/consolidation.go

```diff
@@ -169,23 +221,45 @@ func (c *Consolidation) computeConsolidation(ctx context.Context, node Candidate
 		newNodes[0].Requirements.Add(scheduling.NewRequirement(v1alpha5.LabelCapacityType, v1.NodeSelectorOpIn, v1alpha5.CapacityTypeSpot))
 	}
 
+	c.setTimestamp(validationTimestamp)
 	return Command{
 		nodesToRemove:    []*v1.Node{node.Node},
 		action:           actionReplace,
 		replacementNodes: []*pscheduling.Node{newNodes[0]},
-		createdAt:        c.clock.Now(),
 	}, nil
 }
 
-// validateCommand validates a command for a deprovisioner
-func (c *Consolidation) ValidateCommand(ctx context.Context, nodesToDelete []CandidateNode, cmd Command) (bool, error) {
-	// If we're deleting and there are empty nodes, command is valid.
-	if cmd.action == actionDelete {
-		if c.validateDeleteEmpty(nodesToDelete) {
-			return true, nil
+// setTimestamp sets the given validationTimestamp pointer to the current time. No-op if already set.
+func (c *Consolidation) setTimestamp(validationTimestamp *time.Time) {
```

Comment:
This mutates data within a function which is generally unexpected. I'm more in favor of keeping our functions pure since it communicates that we are actually setting a value. 

Why not just call each time
```go
validationTimestamp = lo.Ternary(validationTimestamp.IsZero(), c.clock.Now(), validationTimestamp)
```